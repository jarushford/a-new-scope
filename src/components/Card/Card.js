import React, { Component } from 'react'
import PropTypes from 'prop-types'
import ResidentsScroller from '../ResidentsScroller/ResidentsScroller'

export default class Card extends Component {
  constructor() {
    super()
    this.state = {
      flipped: false,
      unflipped: false,
      cardObj: null
    }
  }

  componentDidMount() {
    const { cardData, cardType } = this.props
    const cardObj = {
      category: cardData.category,
      favorite: cardData.favorite,
      name: cardData.name,
      type: cardData.type,
      main1Label: '',
      main1: cardData.main1,
      main2Label: '',
      main2: cardData.main2,
      secHeader: '',
      secInfoMain: cardData.secInfoMain,
      secInfoOther: cardData.secInfoOther || ''
    }

    switch (cardType) {
      case 'people':
        cardObj.main1Label = 'Height'
        cardObj.main2Label = 'Weight'
        cardObj.secHeader = 'Homeworld'
        break
      case 'vehicles':
        cardObj.main1Label = 'Model'
        cardObj.main2Label = 'Class'
        cardObj.secHeader = 'Passengers'
        break
      case 'planets':
        cardObj.main1Label = 'Population'
        cardObj.main2Label = 'Climate'
        cardObj.secHeader = 'Residents'
        break
      default:
        break
    }

    this.setState({ cardObj })
  }

  flipCard = (e) => {
    const { flipped } = this.state
    if (e.target.classList.contains('resident-arrow')
      || e.target.classList.contains('favorite-btn')) { return }

    if (flipped === true) {
      this.setState({
        flipped: false,
        unflipped: true
      })
    } else {
      this.setState({
        flipped: true,
        unflipped: false
      })
    }
  }

  toggleFavorite = async (cardObj) => {
    const newCardObj = { ...cardObj, favorite: !cardObj.favorite }
    const { handleStoreData } = this.props
    await this.setState({ cardObj: newCardObj })
    handleStoreData('favorites', newCardObj, newCardObj.favorite, cardObj.category)
    if (this.props.favorite === 'favorites') {
      this.props.updateFavorites()
    }
  }

  render() {
    const { flipped, unflipped, cardObj } = this.state
    const { cardType } = this.props
    let cardObjPeople
    if (cardType === 'people') {
      cardObjPeople = true
    } else {
      cardObjPeople = ''
    }
    if (cardObj === null) {
      return (<div />)
    }
    return (
      <div
        className={`card ${flipped && 'flipped'} ${unflipped && 'unflipped'}`}
        role="presentation"
        onClick={e => this.flipCard(e)}
      >
        <div className="card-front">
          <div className={`image-side-title card-title card-header-${cardType}`}>
            <h1>{cardObj.name}</h1>
            <i
              className={`fas fa-star ${cardObj.favorite && 'favorite'} favorite-btn`}
              role="presentation"
              onClick={() => this.toggleFavorite(cardObj)}
            />
          </div>
          <img className="front-image" src={`./images/${cardObj.name.replace('/', '-')}.jpg`} alt="" />
        </div>
        <div className={` card-title card-header-${cardType}`}>
          <h1>{cardObj.name}</h1>
          <i
            className={`fas fa-star ${cardObj.favorite && 'favorite'} favorite-btn`}
            role="presentation"
            onClick={() => this.toggleFavorite(cardObj)}
          />
        </div>
        <h1 className="card-type">{cardObj.type}</h1>
        <div className="stats-container">
          <div className="stat">
            <h1>{cardObj.main1Label}</h1>
            <p>{`${cardObj.main1} ${cardObjPeople && 'cm'}`}</p>
          </div>
          <div className="div-line" />
          <div className="stat">
            <h1>{cardObj.main2Label}</h1>
            <p>{`${cardObj.main2} ${cardObjPeople && 'kg'}`}</p>
          </div>
        </div>
        <div>
          <div className={`homeworld-title card-header-${cardType}`}>
            <h1 className={`card-header-${cardType}`}>{cardObj.secHeader}</h1>
          </div>
          <ResidentsScroller
            content1={cardObj.secInfoMain}
            content2={cardObj.secInfoOther}
          />
        </div>
        <img className="card-icon" alt="card icon" src={`./images/${cardType}_icon.svg`} />
      </div>
    )
  }
}

Card.propTypes = {
  cardType: PropTypes.string,
  cardData: PropTypes.object.isRequired,
  handleStoreData: PropTypes.func,
  favorite: PropTypes.string,
  updateFavorites: PropTypes.func
}

Card.defaultProps = {
  cardType: '',
  handleStoreData: null
}
