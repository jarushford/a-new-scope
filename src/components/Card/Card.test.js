import React from 'react'
import { shallow } from 'enzyme'
import Card from './Card'
import { O_TRUNC } from 'constants';

describe('Card', () => {
  let wrapper
  const mockData = {
    name: 'Test Planet',
    type: 'Fun Terrrain',
    main1: '200K',
    main2: 'Warm',
    secInfoMain: ['Jim', 'Bob'],
    secInfoOther: 'None',
    favorite: false,
    category: 'planets'
  }
  let formattedData
  const handleStoreData = jest.fn()
  const updateFavorites = jest.fn()
  let favorite = 'planets'

  beforeEach(() => {
    wrapper = shallow(
      <Card
        cardType="planets"
        cardData={mockData}
        handleStoreData={handleStoreData}
        updateFavorites={updateFavorites}
        favorite='favorites'
      />
    )
    formattedData = {
      favorite: false,
      category: 'planets',
      name: 'Test Planet',
      type: 'Fun Terrrain',
      main1Label: 'Population',
      main1: '200K',
      main2Label: 'Climate',
      main2: 'Warm',
      secHeader: 'Residents',
      secInfoMain: ['Jim', 'Bob'],
      secInfoOther: 'None'
    }
  })

  it('Should default flipped to false, and unflipped to false', () => {
    wrapper = shallow(<Card cardType="planets" cardData={mockData} />, { disableLifecycleMethods: true })
    const expected = {
      flipped: false,
      unflipped: false,
      cardObj: null
    }

    expect(wrapper.state()).toEqual(expected)
  })

  it('Should render an empty div if cardObj is null', () => {
    wrapper = shallow(<Card cardType="planets" cardData={mockData} />, { disableLifecycleMethods: true })

    expect(wrapper).toMatchSnapshot()
  })

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('ComponentDidMount', () => {
    it('Should set CardObj to reformatted cardData', () => {
      const expected = {
        flipped: false,
        unflipped: false,
        cardObj: formattedData
      }

      expect(wrapper.state()).toEqual(expected)
    })

    it('Should set CardObj labels to vehicle labels if category is vehicle', () => {
      wrapper = shallow(<Card cardType="vehicles" cardData={mockData} />)
      const expected = {
        favorite: false,
        category: 'planets',
        name: 'Test Planet',
        type: 'Fun Terrrain',
        main1Label: 'Model',
        main1: '200K',
        main2Label: 'Class',
        main2: 'Warm',
        secHeader: 'Passengers',
        secInfoMain: ['Jim', 'Bob'],
        secInfoOther: 'None'
      }

      expect(wrapper.instance().state.cardObj).toEqual(expected)
    })

    it('should set cardObj labels if category is people', () => {
      wrapper = shallow(<Card cardType="people" cardData={mockData} />)
      const expected = {
        favorite: false,
        category: 'planets',
        name: 'Test Planet',
        type: 'Fun Terrrain',
        main1Label: 'Height',
        main1: '200K',
        main2Label: 'Weight',
        main2: 'Warm',
        secHeader: 'Homeworld',
        secInfoMain: ['Jim', 'Bob'],
        secInfoOther: 'None'
      }

      expect(wrapper.instance().state.cardObj).toEqual(expected)
    })

    it('Should not set CardObj labels if there is no Category', () => {
      wrapper = shallow(<Card cardType="" cardData={mockData} />)
      const expected = {
        favorite: false,
        category: 'planets',
        name: 'Test Planet',
        type: 'Fun Terrrain',
        main1Label: '',
        main1: '200K',
        main2Label: '',
        main2: 'Warm',
        secHeader: '',
        secInfoMain: ['Jim', 'Bob'],
        secInfoOther: 'None'
      }
      expect(wrapper.instance().state.cardObj).toEqual(expected)
    })

    it('Should fire the toggleFavorite function when the front favorite button is clicked', () => {
      wrapper.find('.favorite-btn').first().simulate('click')
      expect(wrapper.state().cardObj.favorite).toEqual(true)
    })

    it('Should fire the toggleFavorite function when the back favorite button is clicked', () => {
      wrapper.find('.favorite-btn').at(1).simulate('click')
      expect(wrapper.state().cardObj.favorite).toEqual(true)
    })
  })

  describe('flipCard', () => {
    it('should not setState if the click occured on the favorite button', () => {
      const expected = {
        flipped: false,
        unflipped: false,
        cardObj: formattedData
      }

      wrapper.find('.card').simulate('click', { target: { classList: { contains: () => true } } })

      expect(wrapper.state()).toEqual(expected)
    })

    it('should not setState if the click occured on a scroll button', () => {
      const expected = {
        flipped: false,
        unflipped: false,
        cardObj: formattedData
      }

      wrapper.find('.card').simulate('click', { target: { classList: { contains: () => true } } })

      expect(wrapper.state()).toEqual(expected)
    })

    it('should set flipped to true when unflipped', () => {
      const expected = {
        flipped: true,
        unflipped: false,
        cardObj: formattedData
      }

      wrapper.find('.card').simulate('click', { target: { classList: { contains: () => false } } })

      expect(wrapper.state()).toEqual(expected)
    })

    it('should set flipped to false when flipped', () => {
      const expected = {
        flipped: false,
        unflipped: true,
        cardObj: formattedData
      }

      wrapper.find('.card').simulate('click', { target: { classList: { contains: () => false } } })
      wrapper.find('.card').simulate('click', { target: { classList: { contains: () => false } } })

      expect(wrapper.state()).toEqual(expected)
    })
  })

  describe('ToggleFavorite', () => {
    it('Should toggle the favorite property on the card object in state', () => {
      const expectedData = true

      wrapper.instance().toggleFavorite(mockData)
      expect(wrapper.state().cardObj.favorite).toEqual(expectedData)
    })

    it('Should store the updated card in local storage', () => {
      wrapper.instance().toggleFavorite(mockData)
      expect(handleStoreData).toHaveBeenCalled()
    })

    it('Should run updateCards if the category is favorites', () => {
      wrapper.instance().toggleFavorite(mockData)
      expect(updateFavorites).toHaveBeenCalled()
    })

    
  })
})
