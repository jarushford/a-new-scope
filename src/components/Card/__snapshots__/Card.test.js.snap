// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardData={
      Object {
        "category": "planets",
        "favorite": false,
        "main1": "200K",
        "main2": "Warm",
        "name": "Test Planet",
        "secInfoMain": Array [
          "Jim",
          "Bob",
        ],
        "secInfoOther": "None",
        "type": "Fun Terrrain",
      }
    }
    cardType="planets"
    favorite="favorites"
    handleStoreData={[MockFunction]}
    updateFavorites={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-front"
        >
          <div
            className="image-side-title card-title card-header-planets"
          >
            <h1>
              Test Planet
            </h1>
            <i
              className="fas fa-star false favorite-btn"
              onClick={[Function]}
              role="presentation"
            />
          </div>
          <img
            alt=""
            className="front-image"
            src="./images/Test Planet.jpg"
          />
        </div>,
        <div
          className=" card-title card-header-planets"
        >
          <h1>
            Test Planet
          </h1>
          <i
            className="fas fa-star false favorite-btn"
            onClick={[Function]}
            role="presentation"
          />
        </div>,
        <h1
          className="card-type"
        >
          Fun Terrrain
        </h1>,
        <div
          className="stats-container"
        >
          <div
            className="stat"
          >
            <h1>
              Population
            </h1>
            <p>
              200K 
            </p>
          </div>
          <div
            className="div-line"
          />
          <div
            className="stat"
          >
            <h1>
              Climate
            </h1>
            <p>
              Warm 
            </p>
          </div>
        </div>,
        <div>
          <div
            className="homeworld-title card-header-planets"
          >
            <h1
              className="card-header-planets"
            >
              Residents
            </h1>
          </div>
          <ResidentsScroller
            content1={
              Array [
                "Jim",
                "Bob",
              ]
            }
            content2="None"
          />
        </div>,
        <img
          alt="card icon"
          className="card-icon"
          src="./images/planets_icon.svg"
        />,
      ],
      "className": "card false false",
      "onClick": [Function],
      "role": "presentation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="image-side-title card-title card-header-planets"
            >
              <h1>
                Test Planet
              </h1>
              <i
                className="fas fa-star false favorite-btn"
                onClick={[Function]}
                role="presentation"
              />
            </div>,
            <img
              alt=""
              className="front-image"
              src="./images/Test Planet.jpg"
            />,
          ],
          "className": "card-front",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Test Planet
                </h1>,
                <i
                  className="fas fa-star false favorite-btn"
                  onClick={[Function]}
                  role="presentation"
                />,
              ],
              "className": "image-side-title card-title card-header-planets",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Test Planet",
                },
                "ref": null,
                "rendered": "Test Planet",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-star false favorite-btn",
                  "onClick": [Function],
                  "role": "presentation",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "front-image",
              "src": "./images/Test Planet.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Test Planet
            </h1>,
            <i
              className="fas fa-star false favorite-btn"
              onClick={[Function]}
              role="presentation"
            />,
          ],
          "className": " card-title card-header-planets",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test Planet",
            },
            "ref": null,
            "rendered": "Test Planet",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-star false favorite-btn",
              "onClick": [Function],
              "role": "presentation",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Fun Terrrain",
          "className": "card-type",
        },
        "ref": null,
        "rendered": "Fun Terrrain",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="stat"
            >
              <h1>
                Population
              </h1>
              <p>
                200K 
              </p>
            </div>,
            <div
              className="div-line"
            />,
            <div
              className="stat"
            >
              <h1>
                Climate
              </h1>
              <p>
                Warm 
              </p>
            </div>,
          ],
          "className": "stats-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Population
                </h1>,
                <p>
                  200K 
                </p>,
              ],
              "className": "stat",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Population",
                },
                "ref": null,
                "rendered": "Population",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "200K ",
                },
                "ref": null,
                "rendered": "200K ",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "div-line",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Climate
                </h1>,
                <p>
                  Warm 
                </p>,
              ],
              "className": "stat",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Climate",
                },
                "ref": null,
                "rendered": "Climate",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Warm ",
                },
                "ref": null,
                "rendered": "Warm ",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="homeworld-title card-header-planets"
            >
              <h1
                className="card-header-planets"
              >
                Residents
              </h1>
            </div>,
            <ResidentsScroller
              content1={
                Array [
                  "Jim",
                  "Bob",
                ]
              }
              content2="None"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1
                className="card-header-planets"
              >
                Residents
              </h1>,
              "className": "homeworld-title card-header-planets",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Residents",
                "className": "card-header-planets",
              },
              "ref": null,
              "rendered": "Residents",
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "content1": Array [
                "Jim",
                "Bob",
              ],
              "content2": "None",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "card icon",
          "className": "card-icon",
          "src": "./images/planets_icon.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-front"
          >
            <div
              className="image-side-title card-title card-header-planets"
            >
              <h1>
                Test Planet
              </h1>
              <i
                className="fas fa-star false favorite-btn"
                onClick={[Function]}
                role="presentation"
              />
            </div>
            <img
              alt=""
              className="front-image"
              src="./images/Test Planet.jpg"
            />
          </div>,
          <div
            className=" card-title card-header-planets"
          >
            <h1>
              Test Planet
            </h1>
            <i
              className="fas fa-star false favorite-btn"
              onClick={[Function]}
              role="presentation"
            />
          </div>,
          <h1
            className="card-type"
          >
            Fun Terrrain
          </h1>,
          <div
            className="stats-container"
          >
            <div
              className="stat"
            >
              <h1>
                Population
              </h1>
              <p>
                200K 
              </p>
            </div>
            <div
              className="div-line"
            />
            <div
              className="stat"
            >
              <h1>
                Climate
              </h1>
              <p>
                Warm 
              </p>
            </div>
          </div>,
          <div>
            <div
              className="homeworld-title card-header-planets"
            >
              <h1
                className="card-header-planets"
              >
                Residents
              </h1>
            </div>
            <ResidentsScroller
              content1={
                Array [
                  "Jim",
                  "Bob",
                ]
              }
              content2="None"
            />
          </div>,
          <img
            alt="card icon"
            className="card-icon"
            src="./images/planets_icon.svg"
          />,
        ],
        "className": "card false false",
        "onClick": [Function],
        "role": "presentation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="image-side-title card-title card-header-planets"
              >
                <h1>
                  Test Planet
                </h1>
                <i
                  className="fas fa-star false favorite-btn"
                  onClick={[Function]}
                  role="presentation"
                />
              </div>,
              <img
                alt=""
                className="front-image"
                src="./images/Test Planet.jpg"
              />,
            ],
            "className": "card-front",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Test Planet
                  </h1>,
                  <i
                    className="fas fa-star false favorite-btn"
                    onClick={[Function]}
                    role="presentation"
                  />,
                ],
                "className": "image-side-title card-title card-header-planets",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Test Planet",
                  },
                  "ref": null,
                  "rendered": "Test Planet",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-star false favorite-btn",
                    "onClick": [Function],
                    "role": "presentation",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "className": "front-image",
                "src": "./images/Test Planet.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Test Planet
              </h1>,
              <i
                className="fas fa-star false favorite-btn"
                onClick={[Function]}
                role="presentation"
              />,
            ],
            "className": " card-title card-header-planets",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test Planet",
              },
              "ref": null,
              "rendered": "Test Planet",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-star false favorite-btn",
                "onClick": [Function],
                "role": "presentation",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Fun Terrrain",
            "className": "card-type",
          },
          "ref": null,
          "rendered": "Fun Terrrain",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="stat"
              >
                <h1>
                  Population
                </h1>
                <p>
                  200K 
                </p>
              </div>,
              <div
                className="div-line"
              />,
              <div
                className="stat"
              >
                <h1>
                  Climate
                </h1>
                <p>
                  Warm 
                </p>
              </div>,
            ],
            "className": "stats-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Population
                  </h1>,
                  <p>
                    200K 
                  </p>,
                ],
                "className": "stat",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Population",
                  },
                  "ref": null,
                  "rendered": "Population",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "200K ",
                  },
                  "ref": null,
                  "rendered": "200K ",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "div-line",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Climate
                  </h1>,
                  <p>
                    Warm 
                  </p>,
                ],
                "className": "stat",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Climate",
                  },
                  "ref": null,
                  "rendered": "Climate",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Warm ",
                  },
                  "ref": null,
                  "rendered": "Warm ",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="homeworld-title card-header-planets"
              >
                <h1
                  className="card-header-planets"
                >
                  Residents
                </h1>
              </div>,
              <ResidentsScroller
                content1={
                  Array [
                    "Jim",
                    "Bob",
                  ]
                }
                content2="None"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1
                  className="card-header-planets"
                >
                  Residents
                </h1>,
                "className": "homeworld-title card-header-planets",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Residents",
                  "className": "card-header-planets",
                },
                "ref": null,
                "rendered": "Residents",
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "content1": Array [
                  "Jim",
                  "Bob",
                ],
                "content2": "None",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "card icon",
            "className": "card-icon",
            "src": "./images/planets_icon.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Card Should render an empty div if cardObj is null 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardData={
      Object {
        "category": "planets",
        "favorite": false,
        "main1": "200K",
        "main2": "Warm",
        "name": "Test Planet",
        "secInfoMain": Array [
          "Jim",
          "Bob",
        ],
        "secInfoOther": "None",
        "type": "Fun Terrrain",
      }
    }
    cardType="planets"
    handleStoreData={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
